### PR TITLE
Fix panic on aggregation functions MIN() and COUNT()

### LINKED ISSUE #257: Panic on SELECT with LIMIT min(0)/max(0)
`parser.ParseQuery` panics with the following input and stack trace

```
SELECT * FROM e LIMIT min(0)
SELECT * FROM t OFFSET x(*)
```

<details>
<summary>Stack trace</summary>

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x50ec01]

goroutine 1 [running]:
github.com/genjidb/genji/sql/query/expr.(*MinFunc).Eval(0xc0000d7da0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/runner/work/genji/genji/genji/sql/query/expr/function.go:242 +0x51
github.com/genjidb/genji/sql/parser.selectConfig.ToTree(0x69fb05, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/runner/work/genji/genji/genji/sql/parser/select.go:261 +0x4ed
github.com/genjidb/genji/sql/parser.(*Parser).parseSelectStatement(0xc0001200f0, 0x55, 0x0, 0x0)
	/home/runner/work/genji/genji/genji/sql/parser/select.go:63 +0x44d
github.com/genjidb/genji/sql/parser.(*Parser).ParseStatement(0xc0001200f0, 0x55, 0x0, 0x0, 0x0)
	/home/runner/work/genji/genji/genji/sql/parser/parser.go:91 +0x8a2
github.com/genjidb/genji/sql/parser.(*Parser).ParseQuery(0xc0001200f0, 0x5bd440, 0xc00001c0a8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1a, 0xc00011e080)
	/home/runner/work/genji/genji/genji/sql/parser/parser.go:70 +0x214
github.com/genjidb/genji/sql/parser.ParseQuery(0x5bd440, 0xc00001c0a8, 0xc00011e080, 0x1a, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000043e70, ...)
	/home/runner/work/genji/genji/genji/sql/parser/parser.go:41 +0x117
github.com/genjidb/genji/fuzz/sqlparser.Fuzz(0x7f58e832d000, 0x1a, 0x1a, 0x3)
	/home/runner/work/genji/genji/genji/fuzz/sqlparser/fuzz.go:13 +0xe9
go-fuzz-dep.Main(0xc000043f70, 0x1, 0x1)
	go-fuzz-dep/main.go:36 +0x1ad
main.main()
	github.com/genjidb/genji/fuzz/sqlparser/go.fuzz.main/main.go:15 +0x52
```
</details>

### PR DESCRIPTION
**Pull request by goku321**
This fixes panic when running aggregator functions: MIN() and COUNT().

Fixes #257

### CODE REVIEW

**Review by tie — CHANGES_REQUESTED** [review]
Thanks! Note that we don’t need to change `go.{mod,sum}` since `go-fuzz` dependency is already pinned in `genji/fuzz` module. Adding a test case would also be nice.

File: sql/parser/select_test.go [hunk 1]
```diff
@@ -222,6 +222,8 @@ func TestParserSelect(t *testing.T) {
 				)),
 			false},
 		{"WithOffsetThenLimit", "SELECT * FROM test WHERE age = 10 OFFSET 20 LIMIT 10", nil, true},
+		{"Invalid use of MIN() aggregator", "SELECT * FROM test LIMIT min(0)", nil, true},
+		{"Invalid use of COUNT() aggregator", "SELECT * FROM test OFFSET x(*)", nil, true},
```
Comment (tdakkota) [inline sql/parser/select_test.go:226]: Could you fix and test other aggregation functions (`MAX()`/`AVG()`/etc) too?
Thanks.

Comment (goku321) [reply sql/parser/select_test.go:226]: Sure, I will 👍

Comment (tdakkota) [reply sql/parser/select_test.go:226]: Thanks!

Comment (goku321) [reply sql/parser/select_test.go:226]: I've fixed other aggregator functions as well.

**Review by asdine — APPROVED** [review]
Thanks!

**Pull request merged**